solr/core/src/java/org/apache/solr/handler/component/SearchHandler.java

```diff
@@ -92,6 +93,9 @@ public class SearchHandler extends RequestHandlerBase
 
   protected static final String SHARD_HANDLER_SUFFIX = "[shard]";
 
+  private static final Collection<SolrException.ErrorCode> NONTOLERANT_ERROR_CODES =
+      List.of(SolrException.ErrorCode.BAD_REQUEST);
+
```

Comment:
maybe just make this a Set -- i.e. `Set<> X = Set.of()` -- avoid the extra input. For a list of size 1 it's not going to matter much, but you're doing "set-like things" on this anyway, so there's little purpose served in going broader and making it a Collection.